web/api/v1/api.go

```diff
@@ -333,6 +337,9 @@ func (api *API) options(r *http.Request) apiFuncResult {
 	return apiFuncResult{nil, nil, nil, nil}
 }
 
+type sum struct{}
+type avg struct{}
+
```

Comment:
We keep the declarations of structs with their methods usually, so you can have `sum` just before its implementation and `avg` similarly (in Go you don't need the type to be declared before the usage, you can have it anywhere).